src/ucp/core/ucp_worker.h

```diff
@@ -273,7 +282,8 @@ typedef struct ucp_worker {
     ucp_worker_cm_t                  *cms;                /* Array of CMs, one for each component */
     ucs_mpool_t                      am_mp;               /* Memory pool for AM receives */
     ucs_mpool_t                      reg_mp;              /* Registered memory pool */
-    ucs_mpool_t                      rndv_frag_mp;        /* Memory pool for RNDV fragments */
+    ucp_worker_mpool_hash_t          mpool_hash;          /* Hash table of memory pools */
+    ucs_recursive_spinlock_t         mpool_hash_lock;     /* Lock to safely update memory pools hash table */
```

Comment:
why need lock? worker already has a global lock

Reply:
If multiple threads are sharing the worker using THREAD_MODE_MULTIPLE, don't we have to ensure thread safety in creating memory pools?

Reply:
all UCP worker API calls are already using UCP_WORKER_THREAD_CS_ENTER_CONDITIONAL